### ISSUE TITLE
`export` modifier on `enum` disables `This condition will always return * ts2845`

### ISSUE THREAD

**Issue by eps1lon** [microsoft/TypeScript#63565]
### 🔎 Search Terms

export enum ts2845

### 🕗 Version & Regression Information

5.4 doesn't flag it without the `export` modifier either.

### ⏯ Playground Link

https://www.typescriptlang.org/play/?target=99&module=0&ts=6.0.0-dev.20260416#code/KYDwDg9gTgLgBMAdgVwLZwIIBMCGYbBQDyyMYpAKgJ5jBwDeAUHJgAqsD6rGA4gKIAaZnG78AykIC+jRlmABjADY4odeREQBneDBrAAXJlz5CJMpT0z1W+AGsAloixwAvMN21XLl0bwFipOQw1LQAdBjsXLx8cAA+sb4mAebBeqGifGLCLAD0OSwFhUXFJaUUABb2mnDWWPYw9hpwAO72iopwOIrNOFTVqjDIUIhwAOQwUMjAo6EAFABMABwALACsAJTZcAD8Y4rAAGYwo1uGo1D2AOblxwDcQA

### 💻 Code

```ts
// remove `export` modifier 
export enum AdapterOutputType {
  APP_PAGE,
  PAGES,
}

declare const type: AdapterOutputType

const kind =
  type === AdapterOutputType.APP_PAGE || AdapterOutputType.PAGES
    //                                   ~ should be flagged with: This condition will always return 'true'.(2845)
    ? 'left'
    : 'right';
```

### 🙁 Actual behavior

No typechecking error

### 🙂 Expected behavior

`ts2845` because the condition is always `true`.

**Comment by mkantor** [comment]
`enum`s can be [declaration-merged](https://www.typescriptlang.org/docs/handbook/declaration-merging.html). There could be another module elsewhere in the program which contains code like this:

```ts
export enum AdapterOutputType {
  SOME_OTHER_PAGE = 3
}
```

That'd suddenly make it so the condition may evaluate to `false`. That code could be in an entirely different package not visible to TypeScript, so the analysis is pessimistic.

**Comment by mkantor** [comment]
On second glance I may have misread. The second part of your condition doesn't compare `AdapterOutputType.PAGES` to anything; you're just talking about the fact that `AdapterOutputType.PAGES === 1` which is truthy (and declaration merging won't change that).

**Comment by eps1lon** [comment]
The program would know all the values after declaration merging. How would more values in `AdapterOutputType` change `AdapterOutputType.PAGES === 1`?

**Comment by RyanCavanaugh** [comment]
Repros in 7.0 as well.
